### PR TITLE
 *: add squashes for joining repos with commits, entries and blobs 

### LINKED ISSUE #263: Modify table: commits
The new columns will be the following ones:

commit_hash
--
commit_author_name
commit_author_email
commit_author_when
committer_name
committer_email
committer_when
commit_message
tree_hash
commit_parents
repository_id

### LINKED ISSUE #264: Modify table: blobs
The new columns will be the following ones:

Blobs Table
--
blob_hash
blob_size
blob_content
repository_id

### LINKED ISSUE #268: Modify table: tree_entries
New columns:

Tree_entries Table
--
repository_id
tree_entry_name
blob_hash
tree_hash
tree_entry_mode

Also, tree_entry_name must only contain the name of the specific tree or tree entry, instead of all the concatenated names to reach a specific tree entry.

If the entry is a tree, blob_hash will be null, If the entry is a blob, tree_hash will be null.

### PR DESCRIPTION
**Pull request by erizocosmico**
Depends on #271 

Closes #268 
Closes #263 
Closes #264

### CODE REVIEW

File: integration_test.go [hunk 1]
```diff
@@ -255,11 +255,13 @@ func TestSquashCorrectness(t *testing.T) {
 		t.Run(q, func(t *testing.T) {
 			expected := queryResults(t, engine, pool, q)
 			result := queryResults(t, squashEngine, pool, q)
+			require.Len(t, result, len(expected))
 			require.ElementsMatch(
 				t,
 				expected,
 				result,
 			)
+			// require.False(t, true)
```
Comment (ajnavarro) [inline integration_test.go]: leftover?

**Comment by erizocosmico** [comment]
Yes
On Fri, 11 May 2018 at 17:12, Antonio Navarro Perez <notifications@github.com<mailto:notifications@github.com>> wrote:

@ajnavarro approved this pull request.

________________________________

In integration_test.go<https://github.com/src-d/gitbase/pull/275#discussion_r187643730>:

>                       require.ElementsMatch(
                                t,
                                expected,
                                result,
                        )
+                       // require.False(t, true)


leftover?

**Comment by ajnavarro** [comment]
@erizocosmico rebase, please

**Comment by erizocosmico** [comment]
Done

**Comment by ajnavarro** [comment]
rebase please!

**Comment by ajnavarro** [comment]
you should do an interactive rebase to remove duplicated commit.

**Comment by erizocosmico** [comment]
Fixed

**Comment by ajnavarro** [comment]
@erizocosmico LGTM

**Comment by ajnavarro** [comment]
@erizocosmico rebase and lgtm when you wish.

**Pull request merged**